src/dynamics/mpas/dyn_comp.F90

```diff
@@ -1028,6 +1045,69 @@ subroutine read_inidat(dyn_in)
 
    theta_m(:,1:nCellsSolve) = theta(:,1:nCellsSolve) * (1.0_r8 + Rv_over_Rd * tracers(ixqv,:,1:nCellsSolve))
 
+   ! if (.not. analytic_ic_active()) then  ! scale dry-air mass
+   !   surface_integral = cam_mpas_global_sum_real(areaCell(1:nCellsSolve))
+   !   write(iulog,*) subname//': Cell area test value = ', surface_integral
+   !   test_value = sqrt(surface_integral/(4.0_r8*pi))
+   !   write(iulog,*) subname//': earth radius from area = ', test_value
+   ! end if
+
+   call mpas_pool_get_config(domain_ptr % configs, 'config_scale_dry_air_mass', mpas_scale_dry_air_mass)
+
+   if (mpas_scale_dry_air_mass) then
+
+      allocate( p_top(nCellsSolve), preliminary_dry_surface_pressure(nCellsSolve), pm(plev) )
+      ! (1) calculate pressure at the lid                                                                                                         
+      do i=1, nCellsSolve
+         p_top(i) = p0*(rgas*rho(plev,i)*theta_m(plev,i)/p0)**(cpair/(cpair-rgas))
+         p_top(i) = p_top(i) - gravity*0.5_r8*(zint(plev+1,i)-zint(plev,i))*rho(plev,i)*(1.0_r8+tracers(ixqv,plev,i))
+      end do
+
+      ! (2) integrate dry mass in column to compute                                                                                               
+      do i=1, nCellsSolve
+         preliminary_dry_surface_pressure(i) = 0.0_r8
+         do k=1, plev
+            preliminary_dry_surface_pressure(i) = preliminary_dry_surface_pressure(i) + gravity*(zint(k+1,i)-zint(k,i))*rho(k,i)
+         end do
+      end do
+
+      ! (3) compute average global dry surface pressure                                                                                           
+      preliminary_dry_surface_pressure(1:nCellsSolve) =  preliminary_dry_surface_pressure(1:nCellsSolve)*areaCell(1:nCellsSolve)
+      sphere_surface_area = cam_mpas_global_sum_real(areaCell(1:nCellsSolve))
+      preliminary_dry_surface_pressure = cam_mpas_global_sum_real(preliminary_dry_surface_pressure(1:nCellsSolve))
+      preliminary_avg_dry_surface_pressure = preliminary_avg_dry_surface_pressure/sphere_surface_area
+      write(iulog,*) subname//': initial dry globally avg surface pressure (hPa) = ', preliminary_avg_dry_surface_pressure/100.
+
+      ! (4) scale dry air density                                                                                                                 
+      ! scaling_ratio = target_avg_dry_surface_pressure/preliminary_avg_dry_surface_pressure
+      scaling_ratio = 1.0
+      rho(:,:) = rho(:,:)*scaling_ratio
+
+      ! (5) reset qv to conserve mass                                                                                                             
+      tracers(ixqv,:,1:nCellsSolve) = tracers(ixqv,:,1:nCellsSolve)/scaling_ratio
+
+      ! (6) integrate down the column to compute full pressure given the density and qv                                                           
```

Comment:
@PeterHjortLauritzen Do we have to worry about dry/wet tracers in this dycore.  ie conserve mass of wet tracers vs conserve mixing ration of dry?

Reply:
All mixing ratios in MPAS (incl. those on the initial condition file) are dry so no.